source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/Statements/WholesaleServicesQuerySnippetProvider.cs

```diff
@@ -142,14 +162,19 @@ private string GetActorsForNonTotalAmountsSelection(string table, string sql)
 
         if (_queryParameters is { RequestedForEnergySupplier: false, ChargeOwnerId: null })
         {
-            // The following is sufficient, as the validations ensure that the grid area(s) is/are a non-empty,
-            // finite set of grid areas the charge owner owns.
-            // If this assumption changes, then the following should be changed to a more complex query,
-            // to ensure the charge owner only gets 'is_tax' charges from grid areas they own.
-            sql += $"""
-                    AND ({table}.{DatabricksContract.GetChargeOwnerIdColumnName()} = '{_queryParameters.RequestedForActorNumber}'
-                         OR {table}.{DatabricksContract.GetIsTaxColumnName()} = true)
-                    """;
+             sql += $"""
+                     AND (
+                         -- if requested for actor is SystemOperator, then charge owner must be SystemOperator and it must not be a tax
```

Comment:
```suggestion
                         -- When the SystemOperator requests data, then only data where the charge owner is the SystemOperator is returned, it can not/must not be a tax.
```

Dont know if the tax part is a "must" or it is a "can not be" case.